File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
    m := map[int]int{}
	for i , num := range nums {
		val, ok := m[num]
		if ok{
			return []int{val, i}
		}
		m[target-num] = i
	}
	return []int{}
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        dic = {} # {val: ind}
        for i in range(len(numbers)):
            if numbers[i] in dic:
                return [dic[numbers[i]]+1, i +1]
            dic[target - numbers[i]] = i
